blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -230,7 +227,7 @@ public HttpResponse validateAndCreate(@JsonBody JSONObject request) {
             //Now we know the token is valid. Lets find credential
             StandardUsernamePasswordCredentials githubCredential = CredentialsUtils.findCredential(getId(), StandardUsernamePasswordCredentials.class, new BlueOceanDomainRequirement());
 
-            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, "github", "Github Access Token", authenticatedUser.getId(), accessToken);
+            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, getId(), "Github Access Token", authenticatedUser.getId(), accessToken);
```

Comment:
We'll need to discuss how to handle this on the UI side. At present, the UI calls into an endpoint such as `/blue/rest/organizations/${orgname}/scm/${scmId}` to look for a single credential that's tied to the SCM ID. To support multiple SCM's, we probably need some changes that are non-trivial, e.g.

- User selects "GitHub Enterprise"
- UI makes an API call to see if any "GHE" tokens have been registered. Perhaps the endpoint /blue/rest/organizations/${orgname}/scm/github-enterprise` needs to return multiple tokens?
-- If no tokens, user sees "Add Token" step as reflected in design comp
-- If at least one token, user sees list of tokens registered (including the API URL endpoint), as well as an option to "Add New..."
- If user selects existing token, flow proceeds as normal...
- If user selects "Add New...", user sees same "Add Token" step as reflected in design comp

Does this sound correct, or am I making this more complicated that necessary? @i386